File: src/za/co/adhd_developers/java/questions/Question36.java
package za.co.adhd_developers.java.questions;

import za.co.adhd_developers.java.tools.Question;
import za.co.adhd_developers.java.tools.Utils;

public class Question36 implements Question
{
    private int limit = 1000000;

    private int answer = 0;

    @Override
    public void doWork()
    {
        for (int currNumb = 0; currNumb < this.limit; currNumb++)
        {
            String base10 = String.valueOf(currNumb);

            if (base10.startsWith("0") || base10.endsWith("0"))
            {
                continue;
            }

            if (!Utils.isPalindromic(base10))
            {
                continue;
            }

            String base2 = Long.toString(currNumb, 2);

            if (base2.startsWith("0") || base2.endsWith("0"))
            {
                continue;
            }

            if (!Utils.isPalindromic(base2))
            {
                continue;
            }

            this.answer += currNumb;
        }
    }

    @Override
    public void printAnswer()
    {
        System.out.println("=================================");
        System.out.println("Question 36");
        System.out.println("The sum of all numbers that are palindromic in base 10 and base 2, is: " + this.answer);
        System.out.println("=================================");
    }
}

File: src/za/co/adhd_developers/kotlin/questions/Question36.kt
package za.co.adhd_developers.kotlin.questions

import za.co.adhd_developers.kotlin.Question
import za.co.adhd_developers.kotlin.tools.isPalindromic

class Question36 : Question {

    var answer = 0

    override fun doWork() {
        for (i in 1 until 1000000) {

            if (i.toString().startsWith("0") || i.toString().endsWith("0")) {
                continue
            }

            val base10 = isPalindromic(i.toString())

            if (base10) {
                val base2Str = Integer.toBinaryString(i)

                if (base2Str.toString().startsWith("0") || base2Str.toString().endsWith("0")) {
                    continue
                }

                val base2 = isPalindromic(base2Str)

                if (base2) {
                    this.answer = this.answer.plus(i)
                }
            }
        }
    }

    override fun printAnswer() {
        println("=================================")
        println("Question 36")
        println("The sum of all numbers that are palindromic in base 10 and base 2, is: " + this.answer)
        println("=================================")
    }
}